File: dsa/9.stack_and_queue/1700_Number_of_Students_Unable_to_Eat_Lunch.java
package stack_and_queue;
import java.util.*;

class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        Queue<Integer> studentsQueue = new LinkedList<Integer>();

        for (int i = 0 ; i < students.length; i ++){
            studentsQueue.add(students[i]);
        }

        int curNmStudent = students.length;
        int count = 0;

        int indSandwiches = 0;

        while (indSandwiches < sandwiches.length){
            if (count >= curNmStudent){
                return curNmStudent;
            }

            if (sandwiches[indSandwiches] == studentsQueue.peek()){
                studentsQueue.remove();
                indSandwiches += 1;
                count = 0;
                curNmStudent -= 1;
            }
            else{
                count += 1;
                studentsQueue.add(studentsQueue.remove());
            }
        }
        return 0;
    }
}

File: dsa/9.stack_and_queue/1700_Number_of_Students_Unable_to_Eat_Lunch.py
import typing as t

class Solution:
    def countStudents(self, students: t.List[int], sandwiches: t.List[int]) -> int:
        num_std_sand0: int = 0
        num_std_sand1: int = 0
        num_sand0: int = 0
        num_sand1: int = 0

        for std in students:
            if std == 0:
                num_std_sand0 += 1
            else:
                num_std_sand1 += 1
        
        for sand in sandwiches:
            if sand == 0:
                num_sand0 += 1
            else:
                num_sand1 += 1
        
        print(num_std_sand0, num_sand0, num_std_sand1, num_sand1)
        
        if num_std_sand0 < num_sand0:
            for ind in range(0, len(sandwiches), 1):
                if sandwiches[ind] == 0:
                    if num_std_sand0 == 0:
                        return len(sandwiches) - ind
                    else:
                        num_std_sand0 -= 1
        
        if num_std_sand1 < num_sand1:
            for ind in range(0, len(sandwiches), 1):
                if sandwiches[ind] == 1:
                    if num_std_sand1 == 0:
                        return len(sandwiches) - ind
                    else:
                        num_std_sand1 -= 1

        return 0